### ISSUE TITLE
error creating symlink pointing to non-existing file

### ISSUE THREAD

**Issue by piranna** [gulpjs/vinyl-fs#140]
While [decompressing the Linux kernel](https://github.com/kevva/decompress-tar/pull/5#issuecomment-168183119) I've found that vinyl-fs crashes when the symlink points to a non-existing file (because the tar file didn't finished yet to decompress), so when trying to open the file to set the stats our belowed file descriptors fail. This only happens with latest version on npm, seems the code has changed here on the repository, so I'll copy here the fragment that fails:

```javascript
  function written(err, fd) {
    if (isErrorFatal(err)) {
      return complete(err);
    }

    if (!file.stat) {
      return complete();
    }

    if (typeof fd === 'number') {
      return stat(fd);
    }

    fs.open(writePath, 'r', function(err, fd) {
      if (err) {
        return complete(err);
      }

      stat(fd);
    });
  }
```

When calling to `fs.open()` it tries to open the file that doesn't exists since the symlink is still pointing to nowhere, so it fails.

**Comment by Klowner** [comment]
In master there's a check to avoid running `fs.chmod` when the written item is a symlink, that avoids this problem, doesn't it?

**Comment by piranna** [comment]
I have checkout master and it's working correctly. Could it be possible to do a new release in npm with it?

**Comment by Klowner** [comment]
I'm pretty sure they intend to release 4.0 soon. You know you can use github branches directly in npm, right?

**Comment by contra** [comment]
`npm install gulpjs/gulp#4.0`

**Comment by piranna** [comment]
> You know you can use github branches directly in npm, right?

Yes, I know, but I'm not using vinyl-fs as a direct dependency. I could force that by setting it as a direct dependency, but it's an ugly hack.

**Comment by phated** [comment]
Published as 2.3.0 because I found some new features in the commit history.  Sorry it couldn't be a patch.
